implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java

```diff
@@ -821,6 +855,23 @@ private void processJaxRsMethod(OpenAPIImpl openApi, ClassInfo resourceClass, Me
         }
     }
 
+    /**
+     * Add api response to api responses using the annotation information
+     *
+     * @param apiResponseAnnotation The api response annotation
+     * @param operation the method operation
+     */
+    private void addApiReponseFromAnnotation(AnnotationInstance apiResponseAnnotation, Operation operation) {
+        String responseCode = JandexUtil.stringValue(apiResponseAnnotation, OpenApiConstants.PROP_RESPONSE_CODE);
+        if (responseCode == null) {
+            responseCode = APIResponses.DEFAULT;
+        }
```

Comment:
I know this method is now called from two locations, but I'm wondering if the exception responses should ever be set with "default". It might also be a good idea to skip the exception annotation if the operation already has the response code declared locally on the resource method.

Maybe in the case of thrown exceptions, it should be required that the `responseCode` be set on the `@APIResponse`, else it is ignored. Alternatively, if the thrown exception is one of the known JAX-RS exceptions that specifically relate to a response code (e.g. `javax.ws.rs.NotFoundException`), then a missing `responseCode` can be inferred. That could be a subsequent PR though.

Reply:
@MikeEdgar Thanks for raising that. Let me try to come up with a solution.

Reply:
Hi @MikeEdgar Can you please help to review my changes based on your comment ?

Reply:
@lamtrhieu - the approach you outlined sounds reasonable.